packages/contracts/src/dollar/DirectGovernanceFarmer.sol

```diff
@@ -43,8 +49,11 @@ contract DirectGovernanceFarmer is ReentrancyGuard {
         uint256 amount
     );
 
-
-    constructor(address _manager, address base3Pool, address depositZap) {
+    constructor(
```

Comment:
`onERC1155Received` should check that the operator is authorized, from matches user, and ID exists

Reply:
not sure what it is for,  I am not the one who created that contract and the goal here is to add another deposit function maybe you should create another issue if we have to accept erc1155 tokens

Reply:
K, out of spec for this issue then.